Task_1/Task_1/src/components/to_do_list/List.jsx

```diff
@@ -0,0 +1,106 @@
+import { useState, useEffect } from "react";
+import TaskBar from "./task_bar";
+import moment from "moment";
+
+const List = (props) => {
+  console.log(props.list);
+  const [filter, setfilter] = useState(props.list);
+
+  useEffect(() => {
+    setfilter(props.list);
+  }, [props.list]);
+
+  function setFilterAll() {
+    setfilter(props.list);
+  }
+
+  function is_expired(obj) {
+    let date = moment(obj.date);
+    let current_date = moment().format();
+    let remaning_minutes = date.diff(current_date, "minute");
+    if (remaning_minutes <= 0) {
+      return true;
+    }
+    return false;
+  }
+
+  function setFilterExpiredTask() {
+    let newList = [];
+    props.list.map((obj) => {
+      if (is_expired(obj) && !obj.status) {
+        newList.push(obj);
```

Comment:
if you are using map method then we don't need to use push() method here. what map does is it returns iterated array only. so optimized syntax will be:

newList = props?.list?.map((obj) => obj);

Reply:
Implemented